staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -184,6 +184,8 @@ type Instances interface {
 	InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error)
 	// InstanceShutdownByProviderID returns true if the instance is shutdown in cloudprovider
 	InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error)
+	// InstanceMetadataByProviderID returns the instance's metadata.
+	InstanceMetadataByProviderID(ctx context.Context, providerID string) (*InstanceMetadata, error)
```

Comment:
Once we have this, we should migrate all calls to NodeAddressesByProviderId and InstanceTypeByProviderId to this call.
So can you please add a comment:
// Deprecated: Remove once all calls are migrated to InstanceMetadataByProviderID
to both these methods.

Reply:
I plan to remove these in follow-up work(when node-controller uses InstanceMetadataByProviderID). It's ok to add this comment.

Reply:
Done.